src/main/java/hudson/plugins/gradle/injection/MavenExtensionsDetector.java

```diff
@@ -25,7 +25,10 @@ static Set<MavenExtension> detect(InjectionConfig config, FilePath workspace) th
             MavenExtensions mavenExtensions = MavenExtensions.fromFilePath(extensionsFile);
 
             Set<MavenExtension> knownExtensions = new HashSet<>();
-            if (mavenExtensions.hasExtension(MavenExtension.GRADLE_ENTERPRISE.getCoordinates()) ||
+            if (mavenExtensions.hasExtension(MavenExtension.DEVELOCITY.getCoordinates()) ||
+                mavenExtensions.hasExtension(MavenCoordinates.parseCoordinates(config.getMavenExtensionCustomCoordinates()))) {
+                knownExtensions.add(MavenExtension.DEVELOCITY);
+            } else if (mavenExtensions.hasExtension(MavenExtension.GRADLE_ENTERPRISE.getCoordinates()) ||
                 mavenExtensions.hasExtension(MavenCoordinates.parseCoordinates(config.getMavenExtensionCustomCoordinates()))) {
```

Comment:
🤔 isn't the `mavenExtensions.hasExtension(MavenCoordinates.parseCoordinates(config.getMavenExtensionCustomCoordinates())` redundant ? ie either it's false in the previous condition so it'll be false there, or it's true and then this block will never get executed.